search/constraint.go

```diff
@@ -246,6 +257,37 @@ func (ec *equalConstraint) filter(ctx context.Context, rg parquet.RowGroup, prim
 			continue
 		}
 		// We cannot discard the page through statistics but we might need to read it to see if it has the value
+		readPgs = append(readPgs, pageToRead{pfrom: pfrom, pto: pto, idx: i, off: int(poff), csz: int(pcsz)})
+	}
+
+	// Did not find any pages
+	if len(readPgs) == 0 {
+		return nil, nil
+	}
+
+	dictOff, dictSz := ec.s.LabelsFile().DictionaryPageBounds(rgIdx, col.ColumnIndex)
```

Comment:
Does this logic only applies to equal so we have it here? Not in regex constraint?